File: spine-corona/spine/SlotData.lua

local SlotData = {}
function SlotData.new (name, boneData)
	if not name then error("name cannot be nil", 2) end
	if not boneData then error("boneData cannot be nil", 2) end
	
	local self = {
		name = name,
		boneData = boneData
	}

	function self:setColor (r, g, b, a)
		self.r = r
		self.g = g
		self.b = b
		self.a = a
	end

	self:setColor(255, 255, 255, 255)
	
	return self;
end
return SlotData

File: spine-c/src/spine/SlotData.c
#include <spine/SlotData.h>
#include <spine/util.h>

SlotData* SlotData_create (const char* name, BoneData* boneData) {
	SlotData* self = CALLOC(SlotData, 1)
	MALLOC_STR(self->name, name)
	CAST(BoneData*, self->boneData) = boneData;
	self->r = 1;
	self->g = 1;
	self->b = 1;
	self->a = 1;
	return self;
}

void SlotData_dispose (SlotData* self) {
	FREE(self->name);
	FREE(self->attachmentName);
	FREE(self);
}

void SlotData_setAttachmentName (SlotData* self, const char* attachmentName) {
	FREE(self->attachmentName);
	if (attachmentName)
		MALLOC_STR(self->attachmentName, attachmentName)
	else
		CAST(char*, self->attachmentName) = 0;
}
